software/bootcamp-code/main.py

```diff
@@ -3,17 +3,93 @@
 import motion
 import cv2
 import time
-from enum import Enum
+from enum import Enum # for state machine enumerate
+import asyncio #for referee commands over websockets
+import websockets #before use: pip3.9 install websockets
+import json #for parsing referee commands into python library
+from threading import Thread
 
-# STATE MACHINE: 3 states - FIND_BALL = 0, CENTER_BALL = 1, DRIVE_TO_BALL = 2
+#STATE MACHINE
 class State(Enum):
     FIND_BALL = 0
-    CENTER_BALL = 1
-    DRIVE_TO_BALL = 2
+    MOVE_CENTER_BALL = 1
+    FIND_BASKET = 2
+    THROW_BALL = 3
+    STOP = 4
+    TEST_THROWER = 5
+    FIND_FURTHEST_BASKET = 6
+    DRIVE_TO_OP_BASKET = 7
 
-def main_loop():
-    debug = True
+class BasketColor(Enum):
+    BLUE=0
+    MAGENTA=1
+
+def run_websocket(command_list):
+    asyncio.new_event_loop().run_until_complete(listen_referee(command_list))
+
+# method for listening to referee commands
+async def listen_referee(command_list):
+    ip_addr='192.168.3.28:8222'
+    async with websockets.connect('ws://'+ip_addr) as websocket:
```

Comment:
I suggest wrapping it in a while as well in case when it has a issue, it will try to connect again